### PR TITLE
Update ProjectStructureAndRoles.md

### PR DESCRIPTION
**Pull request by MarkusKuemmerle**

### CODE REVIEW

File: ProjectStructureAndRoles.md [hunk 1]
```diff
@@ -75,17 +70,17 @@ A Project participant may resign by notifying the team mailing list and by sendi
 
 **Working Group participants**
 
-Each Project participant can be part of one or more of the Working Groups of the Project by adding his name to the WG_PARTICIPANTS.MD files. By that he gets the role Working Group participant.
+Each Project participant can be part of one or more of the Working Groups of the Project by subscribing to the mailing list(s). By that he gets the role Working Group participant, the WG_PARTICIPANTS.MD file(s) is/are then also adjusted by the administrators of the Project.
```
Comment (jordonezlucena) [inline ProjectStructureAndRoles.md:74]: By that "he**/she**"

File: ProjectStructureAndRoles.md [hunk 2]
```diff
@@ -6,11 +6,6 @@
...
-- Horizontal: Scalability, Architecture, Lifecycle, Service
-
 Each Sub Project is documented in a separate repository.
 
 Each Sub Project must have a README.MD file (with a description of the Sub Project), a CODEOWNERS and a MAINTAINERS.MD file. A Sub Project should have at least one Maintainer. Ideally a Sub Project is managed by two or more Maintainers, depending on the size and scope of the Sub Project. Here, the responsibilities must be clearly agreed upon between all of the Maintainers. This includes coordinating who is responsible for which issues and pull requests. The Contributors to a Sub Project are recorded by the mechanisms of GitHub in the Sub Project’s repository. Each Contributor, Codeowner and Maintainer should also be listed in the [PARTICIPANTS.MD](./PARTICIPANTS.MD) file in the Governance repository root directory. Each Sub Project should have a license file, a GOVERNANCE.MD file (pointing to the Governance repository) and subdirectories /documentation/API_documentation, /documentation/MeetingMinutes, /documentation/SupportingDocuments, /code/API_code and /code/API_definitions. Writing permission to a Sub Project only codeowners should have.
```
Comment (jordonezlucena) [inline ProjectStructureAndRoles.md:11]: Why "Ideally, a Sub Project is managed by two or more Maintainers". What about, as a general rule, having one single maintainer, and for those cases where the Sub Project gets bigger (and thus difficult to be managed by one single maintainer), open the door for additional maintainers?

File: ProjectStructureAndRoles.md [hunk 3]
```diff
@@ -75,17 +70,17 @@ A Project participant may resign by notifying the team mailing list and by sendi
...
 
 **Codeowners**
 
-Codeowners can merge code and accept pull requests. A Codeowner is responsible for the overall quality and stewardship of the Project. Contributors can get a Codeowner role if they are deeply involved in contributing code, pull request review, and triaging issues in the Sub Project for a minimum of three months and perform an ongoing contribution. Codeowners are decided by the Maintainers and formalized by changing the CODEOWNERS file. Changes in codeownership have to be announced on the CAMARA Project mailing list.
+Codeowners can merge code and accept pull requests. A Codeowner is responsible for the overall quality and stewardship of the Project. Contributors can get a Codeowner role if they are Contributors and deeply involved in contributing code, pull request review, and triaging issues in the Sub Project for a minimum of three months and perform an ongoing contribution. Codeowners are decided by the Maintainers and formalized by changing the CODEOWNERS file. Changes in codeownership have to be announced on the CAMARA Project mailing list.
```
Comment (jordonezlucena) [inline ProjectStructureAndRoles.md:83]: why a minimum of three months?
What about saying " A Codeowner is responsible for the overall quality and stewardship of the Project. Contributors can get a Codeowner role if they are Contributors and deeply involved in contributing code, pull request review, perform an ongoing contribution. Codeowners are decided by the [...]"

File: ProjectStructureAndRoles.md [hunk 4]
```diff
@@ -75,17 +70,17 @@ A Project participant may resign by notifying the team mailing list and by sendi
...
 
-A Working Group participant may resign by removing his name of the WG_PARTICIPANTS.MD file.
+A Working Group participant may resign by unsubscribing from the mailing list(s), the WG_PARTICIPANTS.MD file(s) is/are adjusted by the administrators of the Project.
 
 **Contributors**
 
-Each Project participant can contribute to the Sub Projects of the Project. Contributors to a Sub Project can contribute by creating pull requests. There’s no formal process for becoming or resigning as Contributor. The list of Contributors of a specific Sub Project is maintained by GitHub.
+Each Project participant can contribute to the Sub Projects of the Project. Contributors to a Sub Project can contribute by creating pull requests. To become or resign as Contributor please subscribe / unsubscribe to the mailing list of the Sub Project. The list of Contributors of a specific Sub Project is maintained by the administrators of the Project using GitHub functionality.
```
Comment (jordonezlucena) [inline ProjectStructureAndRoles.md:79]: It is said that "The list of Contributors of a specific Sub Project is maintained by the administrators of the Project using GitHub functionality". Why contributors are maintained by CAMARA administrators rather than Sub Project maintainers?

File: ProjectStructureAndRoles.md [hunk 5]
```diff
@@ -75,17 +70,17 @@ A Project participant may resign by notifying the team mailing list and by sendi
...
 
 **Codeowners**
 
-Codeowners can merge code and accept pull requests. A Codeowner is responsible for the overall quality and stewardship of the Project. Contributors can get a Codeowner role if they are deeply involved in contributing code, pull request review, and triaging issues in the Sub Project for a minimum of three months and perform an ongoing contribution. Codeowners are decided by the Maintainers and formalized by changing the CODEOWNERS file. Changes in codeownership have to be announced on the CAMARA Project mailing list.
+Codeowners can merge code and accept pull requests. A Codeowner is responsible for the overall quality and stewardship of the Project. Contributors can get a Codeowner role if they are Contributors and deeply involved in contributing code, pull request review, and triaging issues in the Sub Project for a minimum of three months and perform an ongoing contribution. Codeowners are decided by the Maintainers and formalized by changing the CODEOWNERS file. Changes in codeownership have to be announced on the CAMARA Project mailing list.
```
Comment (jordonezlucena) [inline ProjectStructureAndRoles.md:84]: why a minimum of three months?
What about saying " A Codeowner is responsible for the overall quality and stewardship of the Project. Contributors can get a Codeowner role if they are Contributors and deeply involved in contributing code, pull request review, perform an ongoing contribution. Codeowners are decided by the [...]"

File: ProjectStructureAndRoles.md (hunk 1 above)
Comment (jordonezlucena) [inline ProjectStructureAndRoles.md:74]: By that " he/she" gets the role [...]

File: ProjectStructureAndRoles.md (hunk 2 above)
Comment (jordonezlucena) [inline ProjectStructureAndRoles.md:11]: Why "Ideally, a Sub Project is managed by two or more Maintainers". What about, as a general rule, having one single maintainer, and for those cases where the Sub Project gets bigger (and thus difficult to be managed by one single maintainer), open the door for additional maintainers?

File: ProjectStructureAndRoles.md (hunk 1 above)
Comment (hdamker) [reply ProjectStructureAndRoles.md:74]: to be consistent and inclusive I propose "they get"

File: ProjectStructureAndRoles.md (hunk 2 above)
Comment (hdamker) [reply ProjectStructureAndRoles.md:11]: My understanding: The group of maintainers decides e.g. about the Codeowners and are more or less the group of people who should be involved in (relevant) pull request reviews. Therefore it makes sense for me to have at least two.

File: ProjectStructureAndRoles.md (hunk 4 above)
Comment (MarkusKuemmerle) [reply ProjectStructureAndRoles.md:79]: That is due to the administration model of GitHub. Only admins of the whole Project !!! can add or remove people

File: ProjectStructureAndRoles.md (hunk 5 above)
Comment (MarkusKuemmerle) [reply ProjectStructureAndRoles.md:84]: The 3 months are standard / good practice for open source projects

File: ProjectStructureAndRoles.md (hunk 1 above)
Comment (MarkusKuemmerle) [reply ProjectStructureAndRoles.md:74]: is changed now

File: ProjectStructureAndRoles.md (hunk 2 above)
Comment (MarkusKuemmerle) [reply ProjectStructureAndRoles.md:11]: I would keep the number of Maintainers open (as it is typically in open source projects). At least 2 makes sense, at best from different companies.

File: ProjectStructureAndRoles.md [hunk 6]
```diff
@@ -75,17 +70,17 @@ A Project participant may resign by notifying the team mailing list and by sendi
...
 
-A Working Group participant may resign by removing his name of the WG_PARTICIPANTS.MD file.
+A Working Group participant may resign by unsubscribing from the mailing list(s), the WG_PARTICIPANTS.MD file(s) is/are adjusted by the administrators of the Project.
 
 **Contributors**
 
-Each Project participant can contribute to the Sub Projects of the Project. Contributors to a Sub Project can contribute by creating pull requests. There’s no formal process for becoming or resigning as Contributor. The list of Contributors of a specific Sub Project is maintained by GitHub.
+Each Project participant can contribute to the Sub Projects of the Project. Contributors to a Sub Project can contribute by creating pull requests. To become or resign as Contributor please subscribe / unsubscribe to the mailing list of the Sub Project. The list of Contributors of a specific Sub Project is maintained by the administrators of the Project using GitHub functionality.
```
Comment (shilpa-padgaonkar) [inline ProjectStructureAndRoles.md:79]: Are these 2 related?
1. Contributors to a Sub Project can contribute by creating pull requests. 
2. To become or resign as Contributor please subscribe / unsubscribe to the mailing list of the Sub Project

What stops a user from creating a PR without subscribing to the mailing list?

Comment (Kevsy) [reply ProjectStructureAndRoles.md:79]: Good point - personally I think anyone creating a PR is a contributor, which is not necessarily the case for everyine subscribed to the mailing list.

**Review by Kevsy — CHANGES_REQUESTED** [review]
All good except clarity need on Shilpa/my comments following line 79

File: ProjectStructureAndRoles.md (hunk 6 above)
Comment (MarkusKuemmerle) [reply ProjectStructureAndRoles.md:79]: If someone new creates a PR he/she will be added to the list of Contributors in the GitHub repo automatically. That's how I've understood GitHub. But then we've manually to add him to the mailing and also to the Project participant list. So that should not be the usual process, also as we don't get a notification of this.
It's better if he/she first subscribes to the list (we get a notification) and then create the PR. That should be the defined process.

Comment (shilpa-padgaonkar) [reply ProjectStructureAndRoles.md:79]: I understand that we would like them to subscribe to mailing list and we can recommend them to do so as a best practice as they will be part of invites/meetings automatically. 
Just that technically this does not make them a contributor, is all.

**Review by shilpa-padgaonkar — APPROVED** [review]
Other than comment https://github.com/camaraproject/Governance/pull/68#discussion_r993562366, LGTM

File: ProjectStructureAndRoles.md (hunk 6 above)
Comment (MarkusKuemmerle) [reply ProjectStructureAndRoles.md:79]: That's correct, if we get the notification from the mailing list we can invite them to be GitHub Contributor. As soon as they've accepted that invitation they can be added within GitHub e.g. as Reviewer to a Pull Request. That should be the standard way doing it.

**Review by Kevsy — APPROVED** [review]

File: ProjectStructureAndRoles.md (hunk 6 above)
Comment (hdamker) [reply ProjectStructureAndRoles.md:79]: I suppose the confusion is coming from the differendes of the terms "member" (of the organization/project), "collaborator" (someone who has accepted the invitation to a repo and has at least read access in the repo) and "contributor "(everyone who has a merged commit in a repo). See here: https://opensource.stackexchange.com/questions/10106/what-is-the-difference-between-contributor-member-and-collaborator 

If I get the proposed roles correct, we have the following rules - please correct if wrong:

* Every contributor in a repo in the project shall be a participant (= member) of the project, subscribing to the mailing lists etc. I'm not sure if this makes sense ... especially for contributors who just leave a fix somewhere. They can stay "Outside Contributors" in GitHub terms. We should not disallow such contributions (but review them for sure).
* Every collaborator in a repo of the project shall be a participant (= member) of the project*. That make sense from governance perspective. Can even be enforced in Github but not allowing the invitation of "Outside Contributors" as collaborators into a repo.
* Every subscriber of the mailing list will get an invitation as collaborator in the respective repo ... ok, if they don't have to accept the invitation and can still receive the infomation from the mailing list (a subscriber to a LF mailing list has not necessarily a GitHub account)
* Every collaborator within a repo has to be subscribed to the respective mailing list. Personally, I wouldn't request that (there are developers who don't like to be on mailing lists), but acceptable as a policy. I understand that the proposed process relies on this policy.

**Review by hdamker — APPROVED** [review]
Changes are ok for me.

We should consider to review the complete document to be more precise regarding "collaborator" vs "contributor" (see https://github.com/camaraproject/Governance/pull/68#discussion_r993839636).

File: ProjectStructureAndRoles.md (hunk 6 above)
Comment (MarkusKuemmerle) [reply ProjectStructureAndRoles.md:79]: I'll take that point into a new issue to clarify the roles Collaborator vs. Contributor. But I would close it here so that we can close that mailing list thing
